src/components/EventDetails/EventDetails.js

```diff
@@ -18,862 +18,1089 @@
-        relatedUserTitle = 'My account';
-      }
-      const relatedUserProfileImage = relatedUser.profileImage || null;
-      // $FlowFixMe
-      const showProfileImage = !relatedUser.type;
+    const formattedValue = formatAmount(value);
 
-      const fee = gasUsed && gasPrice ? Math.round(gasUsed * gasPrice) : 0;
-      const freeTx = isPPNTransaction || tag === SET_SMART_WALLET_ACCOUNT_ENS;
+    const directionIcon = isReceived ? 'received' : 'sent';
+    let directionSymbol = isReceived ? '+' : '-';
 
-      const showFeeBlock = (toMyself || !isReceived) && !isPending && (freeTx || !!fee);
-      let showNote = true;
-      const listSettledAssets = (tag === PAYMENT_NETWORK_TX_SETTLEMENT && !isEmpty(extra));
+    if (formattedValue === '0') {
+      directionSymbol = '';
+    }
 
-      if (tag === PAYMENT_NETWORK_TX_SETTLEMENT
-        || tag === PAYMENT_NETWORK_ACCOUNT_TOPUP
-        || tag === PAYMENT_NETWORK_ACCOUNT_WITHDRAWAL
-      ) {
-        showNote = false;
-      }
+    const isPending = isPendingTransaction(event);
+
+    let eventData: ?EventData = null;
+
+    switch (event.tag) {
+      case PAYMENT_NETWORK_ACCOUNT_DEPLOYMENT:
+        return {
+          name: 'Smart Wallet',
+          itemImageSource: smartWalletIcon,
+          actionTitle: 'Activated',
+          actionSubtitle: this.getFeeLabel(),
+          buttons: [
+            {
+              title: 'Activate Pillar Network',
+              onPress: this.topUpPillarNetwork,
+              secondary: true,
+            },
+            {
+              title: 'Refer friends',
+              onPress: this.referFriends,
+              squarePrimary: true,
+            },
+          ],
+        };
+      case PAYMENT_NETWORK_ACCOUNT_TOPUP:
+        if (activeBlockchainNetwork === BLOCKCHAIN_NETWORK_TYPES.PILLAR_NETWORK) {
+          return {
+            name: 'Pillar Network',
+            itemImageSource: PPNIcon,
+            actionTitle: `+ ${formattedValue} ${event.asset}`,
+            actionColor: this.getColor('positive'),
+            actionSubtitle: 'Top up',
+            buttons: [
+              {
+                title: 'Send',
+                onPress: this.sendSynthetic,
+                secondary: true,
+              },
+              {
+                title: 'Top up more',
+                onPress: this.topUpPillarNetwork,
+                squarePrimary: true,
+              },
+            ],
+          };
+        }
+        return null;
+      case SET_SMART_WALLET_ACCOUNT_ENS:
+        return {
+          name: 'ENS name',
+          itemImageSource: smartWalletIcon,
+          actionTitle: 'Registered',
+          actionSubtitle: event.extra.ensName,
+          buttons: [
+            {
+              title: 'View on the blockchain',
+              onPress: this.viewOnTheBlockchain,
+              secondary: true,
+            },
+          ],
+        };
+      case PAYMENT_NETWORK_ACCOUNT_WITHDRAWAL:
+        return null;
```

Comment:
It means that there is no modal for withdrawal?

Reply:
There is no design for withdrawal modal.